priority_queuee.c

```diff
@@ -0,0 +1,318 @@
+{
+
+while(pQueue->front)
+{
+	pQueue->rear=pQueue->front->next;
+	free(pQueue->front);
+	pQueue->front=pQueue->rear;
+}
+pQueue->size=0;
+
+}
+uint8_t  PriorityQueue_IsEmpty    (PriorityQueue_t *pQueue)
+{
+return (pQueue->front==0);
+
+}
+uint8_t  PriorityQueue_IsFull     (PriorityQueue_t *pQueue)
+{
+	return 0;
+
+}
+QueueEntry PriorityQueue_GetTop (PriorityQueue_t *pQueue)
+{
+	return (pQueue->rear->value);
+}
+
+#elif PQ_TYPE == ARRAY_DEC_PQUEUE
+
+void BubbleSort(QueueEntry Array[] , uint16_t Size)
+{
+	uint16_t i ,j;
+    for( i=0 ; i<Size ; i++ )
+    {
+        for( j=i ; j<Size ; j++)
+        {
+            if(Array[i]<Array[j])
+            {
+            	QueueEntry Temp = Array[i];
+                Array[i] =Array[j];
+                Array[j]=Temp;
+            }
+        }
+    }
+}
+
+void PriorityQueue_CreateQueue(PriorityQueue_t *pQueue)
+{
+	pQueue->Front= 0;
+	pQueue->Size = 0;
+}
+
+void PriorityQueue_EnQueue (PriorityQueue_t *pQueue , QueueEntry Data)
+{
+	pQueue->Array[pQueue->Size++]=Data;
+	BubbleSort(pQueue->Array, pQueue->Size);
+	pQueue->Front=0;
+}
+
+QueueEntry PriorityQueue_DeQueue (PriorityQueue_t *pQueue)
+{ QueueEntry x=pQueue->Array[pQueue->Front];
+	pQueue->Front++;
+	pQueue->Size --;
+	return x;
+}
+
+
+uint8_t PriorityQueue_IsEmpty (PriorityQueue_t *pQueue)
+{
+	return (pQueue->Size==0) ? 1 : 0 ;
+}
+
+
+uint8_t PriorityQueue_IsFull  (PriorityQueue_t *pQueue)
+{
+	return (pQueue->Size==PQ_MAX_SIZE) ? TRUE : FALSE ;
+}
+
+
+uint16_t PriorityQueue_GetQueueSize (PriorityQueue_t *pQueue)
+{
+	return pQueue->Size;
+}
+
+
+QueueEntry PriorityQueue_GetTop (PriorityQueue_t *pQueue)
+{
+	return pQueue->Array[pQueue->Front];
+}
+
+void PriorityQueue_ClearQueue (PriorityQueue_t *pQueue)
+{
+	pQueue->Size=0;
+}
+
+void PriorityQueue_TraverseQueue (PriorityQueue_t *pQueue , void (*pFun)(QueueEntry Data))
+{
+	uint16_t Loc_Counter;
+	for(Loc_Counter=0 ; Loc_Counter<pQueue->Size ; Loc_Counter++)
+	{
+		pFun(pQueue->Array[Loc_Counter]);
+	}
+}
+
+
+#elif PQ_TYPE == ARRAY_INC_PQUEUE
+
+void BubbleSort(QueueEntry Array[] , uint16_t Size)
+{
+	uint16_t i ,j;
+    for( i=0 ; i<Size ; i++ )
+    {
+        for( j=i ; j<Size ; j++)
+        {
+            if(Array[i]>Array[j])
+            {
+            	QueueEntry Temp = Array[i];
+                Array[i] =Array[j];
+                Array[j]=Temp;
+            }
+        }
+    }
+}
+
+void PriorityQueue_CreateQueue(PriorityQueue_t *pQueue)
+{
+	pQueue->Front= 0;
+	pQueue->Size = 0;
+}
+
+void PriorityQueue_EnQueue (PriorityQueue_t *pQueue , QueueEntry Data)
+{
+	pQueue->Array[pQueue->Size++]=Data;
+	BubbleSort(pQueue->Array, pQueue->Size);
+	pQueue->Front=0;
+}
+
+QueueEntry PriorityQueue_DeQueue (PriorityQueue_t *pQueue)
+{ QueueEntry x=pQueue->Array[pQueue->Front];
+	pQueue->Front++;
+	pQueue->Size --;
+	return x;
+}
+
+
+uint8_t PriorityQueue_IsEmpty (PriorityQueue_t *pQueue)
+{
+	return (pQueue->Size==0) ? 1 : 0 ;
```

Comment:
at `STD_Types .h` there is a bool data type with contain `true` or `false` so you can use it instead `1` and `0`